File: golang/q1000/q1023.go
package q1000

import (
	"regexp"
)

func camelMatchPlanA(queries []string, pattern string) []bool {
	p := []byte("[a-z]*")
	s := make([]byte, 0, len(pattern)*7+2)
	s = append(s, '^')
	for i := 0; i < len(pattern); i++ {
		v := pattern[i]
		s = append(s, p...)
		s = append(s, v)

	}
	s = append(s, p...)
	s = append(s, '$')
	r := make([]bool, len(queries))
	reg := regexp.MustCompile(string(s))
	for i, v := range queries {
		r[i] = reg.MatchString(v)
	}
	return r
}

File: hs/src/Q1000/Q1023.hs
module Q1000.Q1023(
    camelMatchPlanA
) where

import Text.Regex.Posix

camelMatchPlanA:: [String] -> String -> [Bool]
camelMatchPlanA queries pattern =
    let
       reg = ("[a-z]*"++concat [v:"[a-z]*"|v<-pattern])
    in
        map (=~ reg) queries
